### ISSUE TITLE
Quantify Productivity in Drasil

### ISSUE THREAD

**Issue by Ant13731** [JacquesCarette/Drasil#2776]
From @smiths in today's meeting, we should try to quantify the claims that Drasil is easier and more productive in the long run.

This issue is to keep track of major changes and uses in Drasil. Then, note the parts that were easy and difficult (perhaps in the Wiki).

- [ ] Drasil Website (notably, on the flexibily and translational skills of Drasil)
- [ ] Chunk Logs (notably, on the reuse of information and the generational aspect of Drasil)

**Comment by Ant13731** [comment]
I finished recording my experience [here](https://github.com/JacquesCarette/Drasil/wiki/Quantifying-Productivity-and-Ease-of-Use) in more of a journal-like style. @smiths, is this sort of what you were going for regarding productivity in Drasil?

**Comment by smiths** [comment]
Yes!  This is great information @Ant13731.  I'm impressed you captured and summarized all that information in a short amount of time.

What your examples point to is the value of redefining productivity.  The new Drasil web-site generator didn't save that many lines of code, but it provides flexibility.  You have a nice list of the things you can do with the new web-site, along with ideas for future improvements.  @JacquesCarette and I wrote a brief paper about the [value of redefining what we mean by productivity](https://arxiv.org/abs/2009.14015).  If we could somehow approximate the equations given in the paper, that would be great, but I think that will take some more thinking.  :smile:

The example with the log files is also a great example.

@szymczdm, the examples from @Ant13731 might help you with your write up on how Drasil adds value to developers.

@tingyuw, when we get to the point of adding lessons to Drasil, that is another opportunity to highlight the value of the Drasil infrastructure.  Hopefully we'll be able to make arguments that adding lessons, which reuse existing knowledge, is more productive than creating a lesson "from scratch."

Another example comes from @bmaclach, who showed that adding a generator for swift required a surprisingly small amount of new code.  That isn't so much about showing the value of Drasil to generate examples, but it does show that expanding Drasil (at least to new programming languages that are consistent with the OO paradigm) is reasonably easy.

In the interest of providing feedback that you can use in the future @Ant13731, I have a few minor comments on your writing:

- "in order to" should usually (maybe always?) be written simply as "to".  This is an example of a [low information content phrase](https://www.webpages.uidaho.edu/range357/extra-refs/empty-words.htm#:~:text=Empty%20Words&text=%22Empty%22%20words%20refer%20to%20Low,of%2C%20to%2C%20and%20with.).
- instead of "a SRS", write "an SRS" because we pronounce SRS as ess ar ess.  That is, we pronounce it as if there is a vowel at the start of the word.

**Comment by Ant13731** [comment]
Ahh okay thanks for the advice @smiths! I always thought "a SRS" looked a little off but I was reasoning that you could read it as the full version. I'll make those changes to the wiki too.
